### PR TITLE
Added dropped Alertmanagers endpoint

### PR DESCRIPTION
**Pull request by Conorbro**
~~Preliminary - waiting on #3544~~ - merged.

Will wait for #3638

### CODE REVIEW

File: notifier/notifier.go [hunk 1]
```diff
@@ -397,6 +398,25 @@ func (n *Notifier) Alertmanagers() []*url.URL {
 	return res
 }
 
+// DroppedAlertmanagers returns a slice of Alertmanager URLs.
+func (n *Notifier) DroppedAlertmanagers() []*url.URL {
+	n.mtx.RLock()
+	amSets := n.alertmanagers
```
Comment (simonpasquier) [inline notifier/notifier.go]: s/alertmanagers/droppedAlertmanagers/?

File: notifier/notifier.go [hunk 2]
```diff
@@ -521,11 +542,12 @@ func (s *alertmanagerSet) Sync(tgs []*targetgroup.Group) {
 	all := []alertmanager{}
 
 	for _, tg := range tgs {
-		ams, err := alertmanagerFromGroup(tg, s.cfg)
+		ams, dams, err := alertmanagerFromGroup(tg, s.cfg)
 		if err != nil {
 			level.Error(s.logger).Log("msg", "Creating discovered Alertmanagers failed", "err", err)
 			continue
 		}
+		s.dams = append(s.dams, dams...)
```
Comment (simonpasquier) [inline notifier/notifier.go]: IIUC `s.dams` grows on every Sync() since it is never reset unlike `s.ams`.

File: notifier/notifier.go (hunk 1 above)
Comment (Conorbro) [reply notifier/notifier.go]: Missed it when resolving merge conflicts, thanks!

**Comment by krasi-georgiev** [comment]
@Conorbro  #3638 is merged so you can revisit this one. 
Might be useful to add some short desc as well.

**Pull request closed without merging**